datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1139,6 +1139,10 @@ void validateRuntimeAndImage(SdxClusterRequest clusterRequest, DetailedEnvironme
         }
     }
 
+    private boolean isImageSpecified(ImageSettingsV4Request imageSettingsV4Request) {
```

Comment:
Can we have this method public somewhere so you won't need to duplicate it's body in both SdxService and ImageCatlaogService?

Reply:
I guess I could move this to an ImageSettingsUtil class but I'm not sure if a null-safe empty string check is worth it TBH

Reply:
I am also not sure whether this refactor would worth it, I am wondering how this was implemented in Env/FreeIPA.

Reply:
I only found this `if` statement on image OS
https://github.com/hortonworks/cloudbreak/blob/ba1f2a1bc7226a1d8ecf689b4ede13021bcf2e65/environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java#L432-L447

Reply:
Yes, this was done for FreeIPA `os` selection.